tests/background/uri-info-test.js

```diff
@@ -1,72 +1,113 @@
 import * as uriInfo from '../../src/background/uri-info';
 import { toResult } from '../promise-util';
 import settings from '../settings.json';
-import { unroll } from '../util';
 
-describe('UriInfo.query', function () {
-  var badgeURL = settings.apiUrl + '/badge';
+describe('UriInfo', () => {
+  const badgeURL = `${settings.apiUrl}/badge`;
+  let fetchStub;
 
-  beforeEach(function () {
-    sinon.stub(window, 'fetch').returns(
-      Promise.resolve(
-        new window.Response('{"total": 1}', {
-          status: 200,
-          headers: {},
-        })
-      )
+  beforeEach(() => {
+    fetchStub = sinon.stub(window, 'fetch').resolves(
+      new window.Response('{"total": 1}', {
+        status: 200,
+        headers: {},
+      })
     );
 
-    sinon.stub(console, 'error');
+    sinon.stub(console, 'warn');
   });
 
-  afterEach(function () {
+  afterEach(() => {
     window.fetch.restore();
-    console.error.restore();
+    console.warn.restore();
   });
 
-  it('sends the correct XMLHttpRequest', function () {
-    return uriInfo.query('tabUrl').then(function () {
-      assert.equal(fetch.callCount, 1);
-      assert.deepEqual(fetch.lastCall.args, [
-        badgeURL + '?uri=tabUrl',
-        { credentials: 'include' },
-      ]);
+  describe('query', () => {
+    it('sends the correct XMLHttpRequest', () => {
```

Comment:
The code doesn't use `XMLHttpRequest` any more. "sends the correct network request" would be better.